diff -ruN fuego-1.1/fuegomain/FuegoMainUtil.cpp fuego-1.1.new/fuegomain/FuegoMainUtil.cpp
--- fuego-1.1/fuegomain/FuegoMainUtil.cpp	2011-01-23 10:35:08.000000000 -0800
+++ fuego-1.1.new/fuegomain/FuegoMainUtil.cpp	2024-02-24 06:39:59.655095945 -0800
@@ -22,7 +22,7 @@
 {
     path normalizedFile = file;
     normalizedFile.normalize();
-    string nativeFile = normalizedFile.native_file_string();
+    string nativeFile = normalizedFile.string();
     SgDebug() << "Loading opening book from '" << nativeFile << "'... ";
     ifstream in(nativeFile.c_str());
     if (! in)
diff -ruN fuego-1.1/go/GoGtpEngine.cpp fuego-1.1.new/go/GoGtpEngine.cpp
--- fuego-1.1/go/GoGtpEngine.cpp	2011-03-05 03:09:01.000000000 -0800
+++ fuego-1.1.new/go/GoGtpEngine.cpp	2024-02-24 06:38:29.261762423 -0800
@@ -378,7 +378,7 @@
     CheckMaxClearBoard();
     if (! m_sentinelFile.empty() && exists(m_sentinelFile))
         throw GtpFailure() << "Detected sentinel file '"
-                           << m_sentinelFile.native_file_string() << "'";
+                           << m_sentinelFile.string() << "'";
     if (Board().MoveNumber() > 0)
         GameFinished();
     Init(Board().Size());
diff -ruN fuego-1.1/gtpengine/GtpEngine.h fuego-1.1.new/gtpengine/GtpEngine.h
--- fuego-1.1/gtpengine/GtpEngine.h	2011-01-23 10:35:02.000000000 -0800
+++ fuego-1.1.new/gtpengine/GtpEngine.h	2024-02-24 06:00:53.436766986 -0800
@@ -44,7 +44,7 @@
     opponent's time.
     Enabling this macro adds a dependency on the Boost.Thread library.
     @see GtpEngine::Ponder() */
-#define GTPENGINE_PONDER 1
+#define GTPENGINE_PONDER 0
 #endif
 
 #ifndef GTPENGINE_INTERRUPT
@@ -53,7 +53,7 @@
     Interrupt() to interrupt a running command.
     Enabling this macro adds a dependency on the Boost.Thread library.
     @see GtpEngine::Interrupt() */
-#define GTPENGINE_INTERRUPT 1
+#define GTPENGINE_INTERRUPT 0
 #endif
 
 //----------------------------------------------------------------------------
diff -ruN fuego-1.1/smartgame/SgHash.h fuego-1.1.new/smartgame/SgHash.h
--- fuego-1.1/smartgame/SgHash.h	2011-01-23 10:34:58.000000000 -0800
+++ fuego-1.1.new/smartgame/SgHash.h	2024-02-24 08:01:46.851747216 -0800
@@ -271,7 +271,7 @@
 }
 
 template<int N>
-std::ostream& operator>>(std::istream& in, const SgHash<N>& hash)
+std::istream& operator>>(std::istream& in, const SgHash<N>& hash)
 {
     std::string str;
     in >> str;
