diff -ruN fuego-1.1/fuegomain/FuegoMain.cpp fuego-1.1.new/fuegomain/FuegoMain.cpp
--- fuego-1.1/fuegomain/FuegoMain.cpp	2011-01-23 10:35:08.000000000 -0800
+++ fuego-1.1.new/fuegomain/FuegoMain.cpp	2024-02-25 20:41:32.547401317 -0800
@@ -62,7 +62,7 @@
 {
     if (programPath == 0)
         return "";
-    return path(programPath, boost::filesystem::native).branch_path();
+    return path(programPath).branch_path();
 }
 
 void Help(po::options_description& desc, ostream& out)
diff -ruN fuego-1.1/fuegomain/FuegoMainUtil.cpp fuego-1.1.new/fuegomain/FuegoMainUtil.cpp
--- fuego-1.1/fuegomain/FuegoMainUtil.cpp	2011-01-23 10:35:08.000000000 -0800
+++ fuego-1.1.new/fuegomain/FuegoMainUtil.cpp	2024-02-25 15:42:04.653394100 -0800
@@ -22,7 +22,7 @@
 {
     path normalizedFile = file;
     normalizedFile.normalize();
-    string nativeFile = normalizedFile.native_file_string();
+    string nativeFile = normalizedFile.string();
     SgDebug() << "Loading opening book from '" << nativeFile << "'... ";
     ifstream in(nativeFile.c_str());
     if (! in)
diff -ruN fuego-1.1/go/GoBoardUtil.cpp fuego-1.1.new/go/GoBoardUtil.cpp
--- fuego-1.1/go/GoBoardUtil.cpp	2011-01-23 10:35:06.000000000 -0800
+++ fuego-1.1.new/go/GoBoardUtil.cpp	2024-02-25 15:42:04.653394100 -0800
@@ -53,9 +53,9 @@
     if (c != SG_EMPTY && ! deadStones.Contains(p))
     {
         if (c == SG_BLACK)
-            ++isBlackAdjacent;
+            isBlackAdjacent = ! isBlackAdjacent;
         else
-            ++isWhiteAdjacent;
+            isWhiteAdjacent = ! isWhiteAdjacent;
         return;
     }
     if (! marker.NewMark(p))
diff -ruN fuego-1.1/go/GoGtpEngine.cpp fuego-1.1.new/go/GoGtpEngine.cpp
--- fuego-1.1/go/GoGtpEngine.cpp	2011-03-05 03:09:01.000000000 -0800
+++ fuego-1.1.new/go/GoGtpEngine.cpp	2024-02-25 15:42:04.653394100 -0800
@@ -378,7 +378,7 @@
     CheckMaxClearBoard();
     if (! m_sentinelFile.empty() && exists(m_sentinelFile))
         throw GtpFailure() << "Detected sentinel file '"
-                           << m_sentinelFile.native_file_string() << "'";
+                           << m_sentinelFile.string() << "'";
     if (Board().MoveNumber() > 0)
         GameFinished();
     Init(Board().Size());
@@ -1608,7 +1608,7 @@
     // Call GoPlayer::Ponder() after 0.2 seconds delay to avoid calls in very
     // short intervals between received commands
     boost::xtime time;
-    boost::xtime_get(&time, boost::TIME_UTC);
+    boost::xtime_get(&time, boost::TIME_UTC_);
     bool aborted = false;
     for (int i = 0; i < 200; ++i)
     {
diff -ruN fuego-1.1/gtpengine/GtpEngine.cpp fuego-1.1.new/gtpengine/GtpEngine.cpp
--- fuego-1.1/gtpengine/GtpEngine.cpp	2011-01-23 10:35:02.000000000 -0800
+++ fuego-1.1.new/gtpengine/GtpEngine.cpp	2024-02-25 15:44:16.623354124 -0800
@@ -118,9 +118,9 @@
 /** Utility functions for Boost.Thread. */
 namespace {
 
-void Notify(mutex& aMutex, condition& aCondition)
+  void Notify(boost::mutex& aMutex, condition& aCondition)
 {
-    mutex::scoped_lock lock(aMutex);
+    boost::mutex::scoped_lock lock(aMutex);
     aCondition.notify_all();
 }
 
@@ -167,15 +167,15 @@
 
     barrier m_threadReady;
 
-    mutex m_startPonderMutex;
+    boost::mutex m_startPonderMutex;
 
-    mutex m_ponderFinishedMutex;
+    boost::mutex m_ponderFinishedMutex;
 
     condition m_startPonder;
 
     condition m_ponderFinished;
 
-    mutex::scoped_lock m_ponderFinishedLock;
+    boost::mutex::scoped_lock m_ponderFinishedLock;
 
     /** The thread to run the ponder function.
         Order dependency: must be constructed as the last member, because the
@@ -190,7 +190,7 @@
 
 void PonderThread::Function::operator()()
 {
-    mutex::scoped_lock lock(m_ponderThread.m_startPonderMutex);
+    boost::mutex::scoped_lock lock(m_ponderThread.m_startPonderMutex);
     m_ponderThread.m_threadReady.wait();
     while (true)
     {
@@ -281,15 +281,15 @@
 
     barrier m_threadReady;
 
-    mutex m_waitCommandMutex;
+    boost::mutex m_waitCommandMutex;
 
     condition m_waitCommand;
 
-    mutex m_commandReceivedMutex;
+    boost::mutex m_commandReceivedMutex;
 
     condition m_commandReceived;
 
-    mutex::scoped_lock m_commandReceivedLock;
+    boost::mutex::scoped_lock m_commandReceivedLock;
 
     /** The thread to run the read command function.
         Order dependency: must be constructed as the last member, because the
@@ -304,7 +304,7 @@
 
 void ReadThread::Function::operator()()
 {
-    mutex::scoped_lock lock(m_readThread.m_waitCommandMutex);
+    boost::mutex::scoped_lock lock(m_readThread.m_waitCommandMutex);
     m_readThread.m_threadReady.wait();
     GtpEngine& engine = m_readThread.m_engine;
     GtpInputStream& in = m_readThread.m_in;
@@ -349,7 +349,7 @@
     {
         cerr << "GtpEngine: sleep " << seconds << '\n';
         xtime time;
-        xtime_get(&time, boost::TIME_UTC);
+        xtime_get(&time, boost::TIME_UTC_);
         time.sec += seconds;
         thread::sleep(time);
         cerr << "GtpEngine: sleep done\n";
diff -ruN fuego-1.1/smartgame/SgHash.h fuego-1.1.new/smartgame/SgHash.h
--- fuego-1.1/smartgame/SgHash.h	2011-01-23 10:34:58.000000000 -0800
+++ fuego-1.1.new/smartgame/SgHash.h	2024-02-25 15:42:04.653394100 -0800
@@ -271,7 +271,7 @@
 }
 
 template<int N>
-std::ostream& operator>>(std::istream& in, const SgHash<N>& hash)
+std::istream& operator>>(std::istream& in, const SgHash<N>& hash)
 {
     std::string str;
     in >> str;
diff -ruN fuego-1.1/smartgame/SgUctSearch.cpp fuego-1.1.new/smartgame/SgUctSearch.cpp
--- fuego-1.1/smartgame/SgUctSearch.cpp	2011-03-13 00:07:38.000000000 -0800
+++ fuego-1.1.new/smartgame/SgUctSearch.cpp	2024-02-25 19:55:13.058329482 -0800
@@ -78,9 +78,9 @@
     return nodesPerTree;
 }
 
-void Notify(mutex& aMutex, condition& aCondition)
+void Notify(boost::mutex& aMutex, condition& aCondition)
 {
-    mutex::scoped_lock lock(aMutex);
+    boost::mutex::scoped_lock lock(aMutex);
     aCondition.notify_all();
 }
 
@@ -191,7 +191,7 @@
     if (DEBUG_THREADS)
         SgDebug() << "SgUctSearch::Thread: starting thread "
                   << m_state->m_threadId << '\n';
-    mutex::scoped_lock lock(m_startPlayMutex);
+    boost::mutex::scoped_lock lock(m_startPlayMutex);
     m_threadReady.wait();
     while (true)
     {
@@ -418,7 +418,7 @@
     {
         auto_ptr<SgUctThreadState> state(
                                       m_threadStateFactory->Create(i, *this));
-        shared_ptr<Thread> thread(new Thread(*this, state));
+        boost::shared_ptr<Thread> thread(new Thread(*this, state));
         m_threads.push_back(thread);
     }
     m_tree.CreateAllocators(m_numberThreads);
diff -ruN fuego-1.1/smartgame/SgUctTree.cpp fuego-1.1.new/smartgame/SgUctTree.cpp
--- fuego-1.1/smartgame/SgUctTree.cpp	2011-01-23 10:34:58.000000000 -0800
+++ fuego-1.1.new/smartgame/SgUctTree.cpp	2024-02-25 15:42:04.653394100 -0800
@@ -455,7 +455,7 @@
     if (m_current->HasChildren())
     {
         SgUctChildIterator* it = new SgUctChildIterator(m_tree, *m_current);
-        m_stack.push(shared_ptr<SgUctChildIterator>(it));
+        m_stack.push(boost::shared_ptr<SgUctChildIterator>(it));
         m_current = &(**it);
         return;
     }
